destination_test.go

```diff
@@ -1,16 +1,140 @@
-package connectorname_test
+// Copyright © 2023 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpcclient
 
 import (
 	"context"
+	"io"
+	"log"
+	"net"
+	"sync"
 	"testing"
 
-	connectorname "github.com/conduitio/conduit-connector-connectorname"
+	pb "github.com/conduitio-labs/conduit-connector-grpc-client/proto/v1"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/golang/mock/gomock"
+	"github.com/matryer/is"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/test/bufconn"
 )
 
 func TestTeardown_NoOpen(t *testing.T) {
-	con := connectorname.NewDestination()
+	is := is.New(t)
+	con := NewDestination()
 	err := con.Teardown(context.Background())
-	if err != nil {
-		t.Errorf("expected no error, got %v", err)
+	is.NoErr(err)
+}
+
+func TestWrite_Success(t *testing.T) {
+	is := is.New(t)
+	dest, ctx := prepareServerAndDestination(t)
+
+	records := []sdk.Record{
```

Comment:
I'd love it if we would assert that the server actually receives the expected records. But I don't want to block the PR because of this, maybe add it as part of the next one? 😉

Reply:
will do